### PR TITLE
Chore: Upgrade hardhat to v2.11

### LINKED ISSUE #359: Upgrade hardhat to v2.11

### PR DESCRIPTION
**Pull request by kyledewy**
## Context
closes #359 

## Changes proposed in this pull request
This PR upgrades hardhat and solidity-coverage to the latest releases. This PR also changed the `typechain` dependency to the hardhat native `@typechain/hardhat`, as well as update `@typechain/truffle-v5`. 

## Test plan
No further tests were added

### CODE REVIEW

File: test/unit/CowOperator/placeOrder.js [hunk 1]
```diff
@@ -679,7 +679,12 @@ describe('placeOrder', function () {
   it('emits an OrderPlaced event', async function () {
     const tx = await swapOperator.placeOrder(contractOrder, orderUID);
     const rcp = await tx.wait();
-
-    expect(rcp.events[2].args.order).to.deep.include.members(Object.values(contractOrder));
+    // Remove duplicate values
+    const rcpOrder = rcp.events[2].args.order.slice(0, 12);
```
Comment (kyledewy) [inline test/unit/CowOperator/placeOrder.js:683]: I don't love this solution. It seems with the upgrade the event receipt has duplicate values  and the BigNumber type is different from parseEther(). Open to suggestions for a cleaner approach.

**Review by shark0der — CHANGES_REQUESTED** [review]
Make sure typechain works, remove custom typechain task, switch to config

File: hardhat.config.js/index.js [hunk 2]
```diff
@@ -21,6 +21,14 @@ const config = {
     apiKey: process.env.ETHERSCAN_API_KEY,
   },
 
+  typechain: {
+    outDir: 'src/types',
```
Comment (shark0der) [inline hardhat.config.js/index.js:25]: `outDir` should be `'types'` (drop the `src/`)

File: hardhat.config.js/index.js [hunk 3]
```diff
@@ -21,6 +21,14 @@ const config = {
...
+    target: 'ethers-v5',
+    alwaysGenerateOverloads: false,
+    externalArtifacts: ['externalArtifacts/*.json'],
+    dontOverrideCompile: true, // defaults to false
```
Comment (shark0der) [inline hardhat.config.js/index.js:29]: - `outDir` should be `'types'` (drop the `src/`)
- we currently don't have external artifacts, remove that line. if we want it - let's do another PR as it's a new feature that's out of scope for this one

**Review by shark0der — APPROVED** [review]

**Pull request merged**